content/blog/how-to-use-dscc-api-and-ansible-to-collect-the-storage-configuration.md

```diff
@@ -0,0 +1,223 @@
+    ansible.builtin.include_tasks:
+      file: DSCC-API-Call.yaml
+
+  - name: Retry the command if status 401
+    vars:
+      request_uri: "/api/v1/storage-systems" 
+    ansible.builtin.include_tasks:
+      file: DSCC-API-401.yaml
+    when: status == '401'
+
+  - name: Set Systems
+    ansible.builtin.set_fact:
+      systems: "{{ response.json['items'] }}"
+    when: status in ['200', '201']
+
+  - name: Initialize Storage system dictionary if not defined
+    ansible.builtin.set_fact:
+      storage_systems: "{{ storage_systems | default({}) }}"
+  - name: Create StorageSystems Dictionary
+    ansible.builtin.set_fact:
+      storage_systems: "{{ storage_systems | combine({item.name: {'id': item.id, 'resourceUri': item.resourceUri}}) }}"
+    with_items: "{{ systems }}"
+
+  - name: Loop Systems
+    vars: 
+    ansible.builtin.include_tasks:
+      file: Loop-Systems.yaml
+    with_dict: "{{storage_systems}}"
+    loop_control:
+      loop_var: my_system
+  
+  - name: Get HostGroups
+    vars:
+      request_uri: "/api/v1/host-initiator-groups"
+    ansible.builtin.include_tasks:
+      file: DSCC-API-Call.yaml    
+  - name: Store the HostGroups
+    ansible.builtin.copy:
+      content: "{{ response.json | to_nice_json }}"
+      dest: "../Outputs/hostGroups.json"
+      mode: '0600'
+    when: response.json is defined
+  
+  - name: Get Hosts
+    ansible.builtin.include_tasks:
+      file: GetAllHosts.yaml
+  - name: Store the Hosts
+    ansible.builtin.copy:
+      content: "{{ response.json | to_nice_json }}"
+      dest: "../Outputs/hosts.json"
+      mode: '0600'
+    when: response.json is defined 
+```
+
+The Loop-Systems.yaml playbook retrieves the storage system details and loops for each system through all the associated links of this array. I.e. you do get a complete capture of the storage array configuration. The captured data is stored in multiple files with the naming structure: **SystemName.associatedLink-Keyname.json.** 
+
+The Ansible playbooks used to capture the system configuration are:
+
+* Capture-Systems.yaml
+
+  * DSCC-API-Call.yaml
+  * DSCC-API-401.yaml
+  * Loop-Systems.yaml
+
+    * Loop-Links.yaml
+    * GetAllSystemVolumes.yaml
+  * GetAllHosts.yaml
+
+In order to keep this blog readable and not code overloaded only pa few of the playbooks used are shown, but all playbooks (and even some more) can be retrieved on Github at: [https://github.com/tbeha/DSCC-Ansible.](https://github.com/tbeha/DSCC-Ansible)
+
+# Conclusion
+
+It is possible to use Ansible playbooks to capture the storage array configuration using the the HPE Data Services Cloud Console REST API and built-in  Ansible functions. Having the storage array captured in one or multiple JSON-files is  leading to an obvious next step: use the captured configuration information to automate the redeployment of a storage array. This is one of my planned next activities. Stay tuned to the [HPE Developer Portal](https://developer.hpe.com/) for more.
```

Comment:
There are two "the" in front of "HPE Data Services Cloud Console REST..." (typo)

Please change "HPE Developer Portal" to "HPE Developer Community blog"